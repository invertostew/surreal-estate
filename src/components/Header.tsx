import React from "react";

import NavBar from "./NavBar";

import "../styles/Header.css";

function Header() {
  return (
    <header className="header">
      <div className="header-logo">
        <img src="#" alt="Surreal Estate" className="header-logo__logo" />
      </div>

      <NavBar />
    </header>
  );
}

export default Header;
