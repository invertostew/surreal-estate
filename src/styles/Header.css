.header {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  padding: 1rem;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 10%), 0 2px 4px -1px rgb(0 0 0 / 6%);
  background-color: #fff;
  position: relative;
  z-index: 1;
}

.header-logo {
  max-height: 80px;
}

.header-logo__logo {
  max-height: 80px;
  width: auto;
}
