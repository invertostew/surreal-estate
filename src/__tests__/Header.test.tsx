import React from "react";
import { render, screen } from "@testing-library/react";

import Header from "../components/Header";

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the logo", () => {
    render(<Header />);

    const logo = screen.getByAltText(/surreal estate/i);

    expect(logo).toBeInTheDocument();
    expect(logo).toHaveClass("header-logo__logo");
  });
});
