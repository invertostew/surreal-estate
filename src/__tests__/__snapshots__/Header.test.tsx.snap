// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header-logo"
    >
      <img
        alt="Surreal Estate"
        class="header-logo__logo"
        src="#"
      />
    </div>
    <nav
      class="navbar"
    >
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-links__item"
        >
          <a
            href="#!"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-links__item"
        >
          <a
            href="#!"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
